src/components/SitePreview.js

```diff
@@ -1,23 +1,31 @@
 // 3rd party libraries - core
-import React from 'react';
+import React, {Component} from 'react';
 import {View, StyleSheet, Dimensions, ImageBackground, TouchableOpacity} from 'react-native';
+import {connect} from 'react-redux';
 import {Icon, Text} from 'react-native-elements';
 import _ from 'lodash';
 
-// 3rd party libraries - additional
+import {returnImageForSiteKey} from '../services/SiteInfoService';
 
 // styles and language
 import {campsite} from '../locale.en';
 
 const {campsite_form: {accessibility_options}} = campsite;
 
-import {featureIconDetails, facilityIconDetails, tokens} from "../constants";
+import {featureIconDetails, facilityIconDetails, tokens, general} from "../constants";
 
 const SCREEN_WIDTH = Dimensions.get('window').width;
 
-const SitePreview = ({navigate, siteToPreview, getSiteDetail, isFavorite, toggleSiteFavorite, currentUser}) => {
+class SitePreview extends Component {
```

Comment:
In order to take advantage of component-level-state, I had to turn all functional components that used `siteImageData` into class based ones.